app/main.py

```diff
@@ -16,4 +16,19 @@
 }
 collection_of_coins = {1, 2, 25}
 
-# write your code here
+def types_variables(variables):
+    sorted_variables = {
+        "mutable": [],
+        "immutable": []
+    }
+
+    for variable in variables:
+        if isinstance(variable, tuple):
+            if all(not isinstance(el, list) for el in variable):
+                sorted_variables["immutable"].append(variable)
+            else:
+                sorted_variables["mutable"].append(variable)
+        elif isinstance(variable, list):
+            sorted_variables["mutable"].append(variable)
+
+    return sorted_variables
```

Comment:
The current implementation only checks for tuples and lists. It misses other mutable types like dictionaries and sets, and immutable types like integers, floats, and strings. Consider adding checks for these types to categorize them correctly.